org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/computation/ElementOrComponentTypeComputer.java

```diff
@@ -90,6 +91,18 @@ protected LightweightTypeReference doVisitUnboundTypeReference(UnboundTypeRefere
 		return null;
 	}
 	
+	@Override
+	protected LightweightTypeReference doVisitCompoundTypeReference(CompoundTypeReference reference) {
+		if (reference.isArray()) {
+		    for (LightweightTypeReference componentType : reference.getMultiTypeComponents()) {
```

Comment:
seems like you could simply call getComponentType on the compound reference, since you've implemented it below.